### DISCUSSION TITLE
Contributions of endogenous to endogenous variables

### DISCUSSION THREAD

**Discussion by pfjulio** [IRIS-Solutions-Team/IRIS-Toolbox#371] [General]
Hi,

Suppose we have a semi-structural model, with several equations like this one:

Y = a * x1 + b * x2 + residual

Equations may be a little more complex than this one, embodying dynamics with lags and an error-correction term. Is it possible, on an equation-by-equation basis, to compute contributions of RHS endogenous variables (the x’s) to the LHS variable (y), plus a residual term?

**Comment by jaromir-benes** [comment]
H- yes, that's what the `Explanatory` equation objects do. See a little
intro into at
https://github.com/IRIS-Solutions-Team/Tutorial-Explanatory-Equations
